dagger-core/src/test/java/io/odpf/dagger/core/source/config/adapter/SourceParquetFilePathsAdapterTest.java

```diff
@@ -0,0 +1,42 @@
+package io.odpf.dagger.core.source.config.adapter;
+
+import com.google.gson.stream.JsonReader;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.io.StringReader;
+
+import static org.junit.Assert.assertArrayEquals;
+
+public class SourceParquetFilePathsAdapterTest {
+    @Test
+    public void shouldDeserializeJsonArrayToStringArray() throws IOException {
+        String parquetFilePathJSONString = "[\"gs://something\", \"gs://anything\"]";
+        JsonReader reader = new JsonReader(new StringReader(parquetFilePathJSONString));
+
+        SourceParquetFilePathsAdapter adapter = new SourceParquetFilePathsAdapter();
+
+        assertArrayEquals(new String[]{"gs://something", "gs://anything"}, adapter.read(reader));
+    }
+
```

Comment:
can we add tests for empty and null inputs?

Reply:
This adapter itself won't be invoked if the config is set to null. Hence, this case is checked by the `StreamConfigValidator` [here](https://github.com/odpf/dagger/pull/152/files/4ab0f5d212e375f422347dde995d50e88548accc#diff-5283f6638c0ff1eacaf3aa667f5ac3e696deec531d30c678306b939cea55d0dfR42) and is being tested [here](https://github.com/odpf/dagger/pull/152/files/4ab0f5d212e375f422347dde995d50e88548accc#diff-393cfe26f1de6214a35dcafd40df596d69b21c2fc3319a7bf86b5b051d6d5a2fR333)


For empty do you mean, what should happen if the config is be set to `[]`  ? If yes, then there is no special handling for it in the code. It will get converted to an empty string array.  I am not sure if I can write a failing test for it.

Reply:
Fixed via commit https://github.com/odpf/dagger/pull/152/commits/c8972fcebe55d361762ab259fd9f462a79b68c74